src/main/java/org/terasology/logic/health/HealthCommands.java

```diff
@@ -72,6 +74,73 @@ public String damageWithType(@Sender EntityRef client, @CommandParam("damageType
             return "Specified damage type does not exist.";
         }
     }
+    public void setAll(DamageResistComponent damageResist,int v){
+        damageResist.damageTypes=new HashMap<>(6);
+        damageResist.damageTypes.put("directDamage",v);
+        damageResist.damageTypes.put("drowningDamage",v);
+        damageResist.damageTypes.put("explosiveDamage",v);
+        damageResist.damageTypes.put("healingDamage",v);
+        damageResist.damageTypes.put("physicalDamage",v);
+        damageResist.damageTypes.put("supportRemovedDamage",v);
+    }
+    @Command(shortDescription = "Damage Resistance", runOnServer = true,
+            requiredPermission = PermissionManager.CHEAT_PERMISSION)
+    public String damageResist(@CommandParam("DamageType") String dType,@CommandParam("Percentage") int value,@Sender EntityRef clientEntity) {
+        if(value>=0&&value<=100) {
+            ClientComponent player = clientEntity.getComponent(ClientComponent.class);
+            DamageResistComponent damageResist = player.character.getComponent(DamageResistComponent.class);
+            String s="ALL";
+            if(damageResist==null) {
+                player.character.addComponent(new DamageResistComponent());
+                damageResist = player.character.getComponent(DamageResistComponent.class);
+                setAll(damageResist,0);
+            }
+            if(dType.equals(s)){
+                    setAll(damageResist,value);
+
+                if(value==0){
+                    player.character.removeComponent(DamageResistComponent.class);
+                }
+            }
+
+            if(damageResist.damageTypes.containsKey(dType)){
+                if(damageResist.damageTypes==null)
+                    damageResist.damageTypes=new HashMap<>(6);
+                damageResist.damageTypes.put(dType,value);
+            }
+            else{
+                if(!dType.equals(s))
+                    return "Not a valid Damage Type";
+            }
+            return "Resistance:"+ value +"% to "+dType;
+        }
+        else{
+            return "accepted values:[0 to 100]";
+        }
+
+
+    }
+
+    @Command(shortDescription = "Damage Immunity",runOnServer = true,
+            requiredPermission = PermissionManager.CHEAT_PERMISSION)
+    public String damageImmune(@CommandParam("DamageType") String dType,@Sender EntityRef clientEntity){
+           return damageResist(dType,100,clientEntity);
+    }
+    @Command(shortDescription = "Check your Resistance", runOnServer = true,
+            requiredPermission = PermissionManager.CHEAT_PERMISSION)
+    public String checkResistance(@Sender EntityRef clientEntity){
+        ClientComponent player= clientEntity.getComponent(ClientComponent.class);
+        DamageResistComponent damageResist= player.character.getComponent(DamageResistComponent.class);
+
+        if(damageResist==null){
```

Comment:
Rather than do a null check you can use a `hasComponent` check and then either indicate the target has no resistances if not or then retrieve the component knowing it exists